govcd/edgegateway.go

```diff
@@ -634,3 +640,53 @@ func (eGW *EdgeGateway) RemoveIpsecVPN() (Task, error) {
 	}
 	return eGW.AddIpsecVPN(ipsecVPNConfig)
 }
+
+// Deletes the edge gateway, returning an error if the vCD call fails.
+// https://code.vmware.com/apis/442/vcloud-director/doc/doc/operations/DELETE-EdgeGateway.html
+func (egw *EdgeGateway) Delete(force bool, recursive bool) (Task, error) {
+	util.Logger.Printf("[TRACE] EdgeGateway.Delete - deleting edge gateway with force: %t, recursive: %t", force, recursive)
+
+	if egw.EdgeGateway.HREF == "" {
+		return Task{}, fmt.Errorf("cannot delete, Object is empty")
+	}
+
+	egwUrl, err := url.ParseRequestURI(egw.EdgeGateway.HREF)
+	if err != nil {
+		return Task{}, fmt.Errorf("error parsing edge gateway url: %s", err)
+	}
+
+	req := egw.client.NewRequest(map[string]string{
```

Comment:
Why ExecuteTaskRequest helper function wasn't used?

Reply:
Because I didn't find a variant of `ExecuteTaskRequest` that also accepts header parameters